src/network/network.cpp

```diff
@@ -1671,62 +1680,55 @@ void Network::Client_Handle_MAP(NetworkConnection& connection, NetworkPacket& pa
 {
 	uint32 size, offset;
 	packet >> size >> offset;
-	if (offset > 0x600000) {
-		// too big
+	int chunksize = packet.size - packet.read;
+	if (chunksize <= 0) {
 		return;
-	} else {
-		int chunksize = packet.size - packet.read;
-		if (chunksize <= 0) {
-			return;
-		}
-		if (offset + chunksize > chunk_buffer.size()) {
-			chunk_buffer.resize(offset + chunksize);
-		}
-		char status[256];
-		sprintf(status, "Downloading map ... (%u / %u)", (offset + chunksize) / 1000, size / 1000);
-		window_network_status_open(status);
-		memcpy(&chunk_buffer[offset], (void*)packet.Read(chunksize), chunksize);
-		if (offset + chunksize == size) {
-			window_network_status_close();
-			bool has_to_free = false;
-			unsigned char *data = &chunk_buffer[0];
-			size_t data_size = size;
-			// zlib-compressed
-			if (strcmp("open2_sv6_zlib", (char *)&chunk_buffer[0]) == 0)
-			{
-				log_warning("Received zlib-compressed sv6 map");
-				has_to_free = true;
-				size_t header_len = strlen("open2_sv6_zlib") + 1;
-				data = util_zlib_inflate(&chunk_buffer[header_len], size - header_len, &data_size);
-				if (data == NULL)
-				{
-					log_warning("Failed to decompress data sent from server.");
-					return;
-				}
-			} else {
-				log_warning("Assuming received map is in plain sv6 format");
-			}
-
-			SDL_RWops* rw = SDL_RWFromMem(data, data_size);
-			if (game_load_network(rw)) {
-				game_load_init();
-				game_command_queue.clear();
-				server_tick = RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_TICKS, uint32);
-				server_srand0_tick = 0;
-				// window_network_status_open("Loaded new map from network");
-				_desynchronised = false;
-			}
-			else
-			{
-				//Something went wrong, game is not loaded. Return to main screen.
-				game_do_command(0, GAME_COMMAND_FLAG_APPLY, 0, 0, GAME_COMMAND_LOAD_OR_QUIT, 1, 0);
-			}
-
-			SDL_RWclose(rw);
-			if (has_to_free)
+	}
+	if (offset + chunksize > chunk_buffer.size()) {
+		chunk_buffer.resize(offset + chunksize);
+	}
+	char status[256];
+	sprintf(status, "Downloading map ... (%u / %u)", (offset + chunksize) / 1000, size / 1000);
+	window_network_status_open(status);
+	memcpy(&chunk_buffer[offset], (void*)packet.Read(chunksize), chunksize);
+	if (offset + chunksize == size) {
+		window_network_status_close();
+		bool has_to_free = false;
+		unsigned char *data = &chunk_buffer[0];
+		size_t data_size = size;
+		// zlib-compressed
+		if (strcmp("open2_sv6_zlib", (char *)&chunk_buffer[0]) == 0)
+		{
+			log_warning("Received zlib-compressed sv6 map");
```

Comment:
Yes, indeed. I originally left it by mistake here while doing some testing and forgot to switch it back to `log_verbose`.